### PR TITLE
[AMORO-1769] Implement optimizer metric collection

### LINKED ISSUE #1769: [Subtask]: Implement optimizer-group metric collection
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### Description

Implement optimizer metric collection.

Design: https://docs.google.com/document/d/1WBHPkkR2bZBX-nGcoEoCTbdGrMZ1AyrPxdxtf91iXo0/edit#heading=h.4boepjbyxjk7
### Parent issue

#1528 

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhongqishang**
## Why are the changes needed?

Close #1769 .

## Brief change log

- Implement optimizer metric collection 

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java [hunk 1]
```diff
@@ -19,6 +19,8 @@
 package com.netease.arctic.server;
 
 import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
```
Comment (baiyangtx) [inline ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java:23]: Please use shaded guava class. 
```suggestion
import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
import org.apache.iceberg.relocated.com.google.common.collect.Maps;
import org.apache.iceberg.relocated.com.google.common.util.concurrent.ThreadFactoryBuilder;
```

File: ams/server/src/main/java/com/netease/arctic/server/metrics/OptimizerReport.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+package com.netease.arctic.server.metrics;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.RatioGauge;
+import com.netease.arctic.ams.api.metrics.MetricReport;
+import com.netease.arctic.ams.api.metrics.TaggedMetrics;
+
+public class OptimizerReport implements MetricReport {
```
Comment (baiyangtx) [inline ams/server/src/main/java/com/netease/arctic/server/metrics/OptimizerReport.java:27]: MetricReport is refactor to MetricsContent.

**Review by baiyangtx — COMMENTED** [review]
There is a conflict with master branch.  could you please fix that?

**Comment by zhongqishang** [comment]
@baiyangtx Thanks for your reviews. I will resolve the conflicts later, and some new changes will be adapted also.

File: ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java [hunk 3]
```diff
@@ -377,4 +397,37 @@ public void run() {
...
+        optimizerReport.optimizerOccupyTime().inc(tableRuntime.getQuotaTime());
+      });
+
+      // Calculate the percentage of time taken by executing tasks for each group
+      // in relation to the total time taken by the optimizer to run.
+      groupCounters.forEach((group, report) -> {
+        int totalThreadCount = listOptimizers(group).stream().mapToInt(Resource::getThreadCount).sum();
+        report.optimizerThreadCount().inc(totalThreadCount);
+        report.optimizerOccupyInterval()
+            .inc(Math.min(System.currentTimeMillis() - startTime, ArcticServiceConstants.QUOTA_LOOK_BACK_TIME));
```
Comment (hameizi) [inline ams/server/src/main/java/com/netease/arctic/server/DefaultOptimizingService.java:428]: The current calculation logic for quotaTime in the process is based on the time spent on optimizing within the last hour. Therefore, the denominator should always be one hour, irrespective of the metric's statistical period.

**Comment by hameizi** [comment]
Also, there is a conflict with master branch.

**Review by hameizi — APPROVED** [review]
LGTM

**Pull request closed without merging**